app/models/login_activity.rb

```diff
@@ -0,0 +1,3 @@
+class LoginActivity < ActiveRecord::Base
+  belongs_to :user, polymorphic: true
```

Comment:
Will it complain when there's a failed login and no user? Might need `optional: true`

Reply:
Thanks, I'll add it.  We're still on Rails 4, but 5 is coming.

Reply:
Oddly, `optional: true` prevents the gem from working.  For now I'm leaving it as described in the docs. https://github.com/ankane/authtrail